lib/validate.js

```diff
@@ -1,35 +1,66 @@
 
-const _ = require('ramda');
+const _           = require('ramda');
+const PropsCheck  = require('props-check')
+const Helper      = require('./helper')
 
+const SPEC = {
+  io        : 'Function'
+, transform : 'Function'
+, awesomize : 'Function'
+}
 
-const throwMiddlewareNotFunction = (index) => {
-  throw TypeError(`All middleware given must be functions - index: ${index}`);
-};
+// @TODO Validate middleware objects
 
-const testMiddlewareArray = _.curry((index, mw) => {
-  try {
-    return testMiddleware(mw);
+// const throwMiddlewareNotFunction = (index) => {
```

Comment:
that validate stuff was all there before but unused and untested. I commented them out so they wouldn't mess with test coverage. There's another ticket open to integrate these validations into the middleware